src/main/java/seedu/hirehub/logic/LogicManager.java

```diff
@@ -68,24 +59,16 @@ public CommandResult execute(String commandText) throws CommandException, ParseE
         case NORMAL:
             command = addressBookParser.parseCommand(commandText);
             break;
-        case CLEARCONFIRM:
-            command = clearConfirmationStageParser.parseCommand(commandText);
-            break;
-        case DELETECONFIRM:
-            command = deleteConfirmationStageParser.parseCommand(commandText);
-            break;
-        case DELETEAPPLICATIONCONFIRM:
-            command = deleteApplicationConfirmationStageParser.parseCommand(commandText);
-            break;
-        case DELETEJOBCONFIRM:
-            command = deleteJobConfirmationStageParser.parseCommand(commandText);
+        case CONFIRM:
+            command = confirmationStageParser.parseCommand(commandText);
             break;
         default:
             break;
         }
         assert (command != null);
         commandResult = command.execute(model);
         state = commandResult.getCommandBoxState();
+        confirmationStageParser.setNextCommands(command);
```

Comment:
its a little weird that the parser has attributes that stores the next commands to be executed right... is there a nicer way to go about this? I cant think of any but if you are able to that'd be great